numpy/core/src/multiarray/buffer.c

```diff
@@ -248,14 +260,17 @@ _buffer_format_string(PyArray_Descr *descr, _tmp_string_t *str,
             /* Insert padding manually */
             if (*offset > new_offset) {
                 PyErr_SetString(PyExc_RuntimeError,
-                                "This should never happen: Invalid offset in "
-                                "buffer format string generation. Please "
-                                "report a bug to the Numpy developers.");
+                                "The buffer interface does not support "
+                                "overlapping fields");
```

Comment:
I'm a little confused by PEP3118, as it:

* Seems to be designed to standardize some numpy behaviour for interoperability
* Causes discussions in the python bugtracker that refer to numpy as the canonical implementation
* Is not actually sufficient to represent either all C types or all numpy types